### PR TITLE
refactor: add module, style fixes

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

Adds the core module and removes the deprecated QBCore export and edits a variable to follow the camelCase rule

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

**Review by Manason — APPROVED** [review]

File: fxmanifest.lua [hunk 1]
```diff
@@ -1,16 +1,21 @@
 fx_version 'cerulean'
 game 'gta5'
 
-description 'vSyncRevamped'
+description 'https://github.com/Qbox-project/qbx-weathersync'
```
Comment (mockdot) [inline fxmanifest.lua]: repository should be used for the github repository, why use description?

Comment (Manason) [reply fxmanifest.lua]: is repository a entry in fxmanifests? Not on the docs https://docs.fivem.net/docs/scripting-reference/resource-manifest/resource-manifest/

Comment (BerkieBb) [reply fxmanifest.lua]: It's not a default entry but ox uses it, so it shall be stolen

**Review by BerkieBb — APPROVED** [review]

**Comment by BerkieBb** [comment]
One thing I would mention is to make an enum out of currentWeather, so we don't need those big if statements to convert the number to a string of weather type

File: fxmanifest.lua (hunk 1 above)
Comment (mockdot) [reply fxmanifest.lua]: Yup, got the idea from ox, I think it's nicer to see the description of the resource and be able to look at the manifest for the github repo

File: fxmanifest.lua [hunk 2]
```diff
@@ -1,7 +1,7 @@
 fx_version 'cerulean'
 game 'gta5'
 
-description 'Weathersync for Qbox'
+description 'QBX-Weathersync'
```
Comment (Manason) [inline fxmanifest.lua]:
```suggestion
description 'Syncs weather and time of day between players'
```

**Review by Manason — APPROVED** [review]

**Pull request merged**